templates/data-sources/file.md.tmpl

```diff
@@ -0,0 +1,36 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "system_file | Data Source | terraform-provider-system"
+name: "system_file"
+type: "Data Source"
+subcategory: ""
+description: |-
+  system_file retrieves information about a file on the remote system.
+---
+
+# Data Source: system_file
+
+`system_file` retrieves information about a file on the remote system.
+
+
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `path` (String) Path to the file. Must be an absolute path.
+
+### Read-Only
+
+- `basename` (String) Base name of the file. Returns the last element of path. Example: Given the attribute `path` is `/path/to/file.txt`, the `basename` is `file.txt`.
+- `content` (String) Content of the file. Only recommended for small text-based payloads such as configuration files etc. In a terraform plan,  The content will be stored in plain-text in the terraform state. Mutually exclusive with attributes `content_sensitive` and `source`.
+- `content_sensitive` (String, Sensitive) Content of the file similar to `content` attribute but with enabled sensitive flag. Prefer `content_sensitive` to `content` to avoid leak of the content in the terraform log output. Mutually exclusive with attributes `content` and `source`.
+- `gid` (Number) ID of the group that owns the file
+- `group` (String) Name of the group that owns the file
+- `id` (String) ID of the file
+- `md5sum` (String) MD5 checksum of the remote file contents on the system in base64 encoding.
+- `mode` (String) Permissions of the file in octal format like `755`. Defaults to the umask of the system.
+- `uid` (Number) ID of the user who owns the file
+- `user` (String) Name of the user who owns the file
+
```

Comment:
The provider documentation of attributes is auto-generated from the schema implementation using [`tfplugindocs`](https://github.com/hashicorp/terraform-plugin-docs). `tfplugindocs` renders each `.md.tmpl` to a corresponding `.md`. After local changes to a `.md.tmpl` template, `tfplugindocs` must be run *before* committing. The resulting `.md` must be part of the commit. I certainly, owe a contribution documentation for this project.

Here is what you could do:

Check `templates/data-sources/command.md.tmpl` as an example.

Apply the suggestion in the template file.

Generate the documentation: Either a) run [`task docs-generate`](https://github.com/neuspaces/terraform-provider-system/blob/main/Taskfile.yml#L107) if you have [task](https://taskfile.dev/) installed
 or b) run `go run github.com/hashicorp/terraform-plugin-docs/cmd/tfplugindocs generate` in the project root. You *do not need* to install tfplugindocs before. As a result, the attributes in `docs/data-sources/file.md` should be perfectly in sync with your Schema implementation.

```suggestion
{{ .SchemaMarkdown | trimspace }}
```